Web Inspector: Incremental in SearchSidebarPanel

francescorn opened the pull request:
#### f02cdaa8457cc2c3a780d013710c82ecbbb4a9df
<pre>
Web Inspector: Incremental in SearchSidebarPanel
<a href="https://bugs.webkit.org/show_bug.cgi?id=260047">https://bugs.webkit.org/show_bug.cgi?id=260047</a>

Reviewed by Devin Rousso.

Replace &quot;search&quot; with &quot;change&quot; to fix search functionality

* Source/WebInspectorUI/UserInterface/Views/SearchSidebarPanel.js:
(WI.SearchSidebarPanel):

Canonical link: <a href="https://commits.webkit.org/266820@main">https://commits.webkit.org/266820@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/10f1eef8fab0aff8d9fe7b17805069f5e935d02a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14838 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15142 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15495 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16586 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13965 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17663 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15243 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/16604 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15018 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15506 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12601 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17319 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12785 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13386 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20360 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13864 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13554 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16790 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14103 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11916 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13393 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3585 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17725 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13946 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/SearchSidebarPanel.js:
```diff
@@ -42,12 +42,17 @@ WI.SearchSidebarPanel = class SearchSidebarPanel extends WI.NavigationSidebarPan
         this._inputElement.setAttribute("placeholder", WI.UIString("Search Resource Content"));
-
+        this._handleSearchInputThrottler = new Throttler(this._searchFieldInput.bind(this), 250);
+        this._inputElement.addEventListener("input", (event) => {
```
so i think we actually should use `"change"`/`"keydown"` instead as we don't really want to `performSearch` until the developer presses the enter key since it's very expensive to search across the entire DOM and every resource and etc.

afaict from reading the C++, it seems like because the `incremental` attribute is _not_ set, `"search"` should only get fired when full changes occur (e.g. the developer hits enter)

as such, i think all we need to do for this PR is just this
```diff
-this._inputElement.addEventListener("search", this._searchFieldChanged.bind(this));
+this._inputElement.addEventListener("change", this._searchFieldChanged.bind(this));
```

francescorn replied:
I understand now, changed.

francescorn committed:
Web Inspector: Incremental in SearchSidebarPanel
https://bugs.webkit.org/show_bug.cgi?id=260047

Reviewed by Devin Rousso.

Replace "search" with "change" to fix search functionality

* Source/WebInspectorUI/UserInterface/Views/SearchSidebarPanel.js:
(WI.SearchSidebarPanel):

Canonical link: https://commits.webkit.org/266820@main
